### PR TITLE
Redesigning evaluator tests

### PR DESCRIPTION
**Pull request by ibralibya**

### CODE REVIEW

File: tests/evaluator_tests.py [hunk 1]
```diff
@@ -82,35 +84,36 @@ def runTest(self):
...
 
-        # removing all top hits, should yield ndcg of 0 as number of recs is 1.
-        for i in range(0, self.users):
-            evaluator.ratings[i, self.predictions[i].nonzero()[0]] = 0
-        ndcg = evaluator.calculate_ndcg(self.n_recommendations, self.predictions,
-                                        self.ratings_matrix, self.test_data)
+        recs_at_x = []
 
-        self.assertEqual(0.0, ndcg)
+        for user in range(self.ratings_matrix.shape[0]):
```
Comment (mostafa-mahmoud) [inline tests/evaluator_tests.py]: Can you use self.users? Also better names for r_at_x

File: tests/evaluator_tests.py [hunk 2]
```diff
@@ -74,6 +74,8 @@ def runTest(self):
         self.assertTrue(numpy.all((test1 * test3) == 0))
 
         evaluator = Evaluator(self.ratings_matrix)
+        print(self.ratings_matrix)
+        print(self.rounded_predictions)
```
Comment (mostafa-mahmoud) [inline tests/evaluator_tests.py]: Remove debugging prints

File: tests/evaluator_tests.py [hunk 3]
```diff
@@ -82,35 +84,36 @@ def runTest(self):
...
-        # recall should be 0.5 by definition
-        for i in range(0, self.users, 2):
-            evaluator.ratings[i, self.predictions[i].nonzero()[0]] = 0
+        # # If we modify all the top predictions for half the users,
+        # # recall should be 0.5 by definition
+        # for i in range(0, self.users, 2):
+        #     evaluator.ratings[i, self.predictions[i].nonzero()[0]] = 0
         recall_at_x = evaluator.recall_at_x(self.n_recommendations, self.predictions,
                                             self.ratings_matrix, self.rounded_predictions)
-        self.assertEqual(0.5, recall_at_x)
```
Comment (mostafa-mahmoud) [inline tests/evaluator_tests.py]: Why did you remove this test?

File: tests/evaluator_tests.py [hunk 4]
```diff
@@ -82,35 +84,36 @@ def runTest(self):
...
+                                             self.test_data, self.rounded_predictions)
 
-        # removing all top hits, should yield ndcg of 0 as number of recs is 1.
-        for i in range(0, self.users):
-            evaluator.ratings[i, self.predictions[i].nonzero()[0]] = 0
-        ndcg = evaluator.calculate_ndcg(self.n_recommendations, self.predictions,
-                                        self.ratings_matrix, self.test_data)
+        recs_at_x = []
 
-        self.assertEqual(0.0, ndcg)
```
Comment (mostafa-mahmoud) [inline tests/evaluator_tests.py]: Why did you remove this test?

**Comment by ibralibya** [comment]
Issues resolved

File: tests/evaluator_tests.py [hunk 5]
```diff
@@ -82,35 +82,34 @@ def runTest(self):
...
-        self.setUp()
-        evaluator.ratings[:] = self.ratings_matrix
+        test_recall_at_x = []
 
-        # removing all top hits, should yield ndcg of 0 as number of recs is 1.
-        for i in range(0, self.users):
-            evaluator.ratings[i, self.predictions[i].nonzero()[0]] = 0
-        ndcg = evaluator.calculate_ndcg(self.n_recommendations, self.predictions,
-                                        self.ratings_matrix, self.test_data)
+        for user in range(self.ratings_matrix.shape[0]):
```
Comment (mostafa-mahmoud) [inline tests/evaluator_tests.py]: Can you rename self.ratings_matrix.shape[0]? it is unclear

File: tests/evaluator_tests.py [hunk 6]
```diff
@@ -82,35 +82,34 @@ def runTest(self):
                 for col in range(self.documents):
                     self.assertEqual(self.rounded_predictions[row, col], self.ratings_matrix[row, col])
 
-        # If we modify all the top predictions for half the users,
-        # recall should be 0.5 by definition
-        for i in range(0, self.users, 2):
-            evaluator.ratings[i, self.predictions[i].nonzero()[0]] = 0
         recall_at_x = evaluator.recall_at_x(self.n_recommendations, self.predictions,
                                             self.ratings_matrix, self.rounded_predictions)
-        self.assertEqual(0.5, recall_at_x)
```
Comment (mostafa-mahmoud) [inline tests/evaluator_tests.py]: Where is the corresponding test?

File: tests/evaluator_tests.py [hunk 7]
```diff
@@ -82,35 +82,34 @@ def runTest(self):
...
         mrr = []
         for i in range(self.users):
             mrr.append(evaluator.calculate_mrr(self.n_recommendations, self.predictions,
-                                               self.rounded_predictions, self.test_data))
+                                               self.test_data, self.rounded_predictions))
             evaluator.ratings[i, (numpy.argmax(self.predictions[i], axis=0))] = 0
             if i > 1:
                 self.assertLessEqual(mrr[i], mrr[i-1])
+
+        evaluator = Evaluator(self.ratings_matrix)
```
Comment (mostafa-mahmoud) [inline tests/evaluator_tests.py]: Is it necessary to create a new one?

Comment (ibralibya) [inline tests/evaluator_tests.py]: yes since the values are manipulated in another test

File: tests/evaluator_tests.py (hunk 6 above)
Comment (ibralibya) [inline tests/evaluator_tests.py]: test is no longer viable

**Comment by mostafa-mahmoud** [comment]
Can you resolve the conflict with lib/collaborative_filtering
